// tceas-utils/entities/appointment.js
"use strict";

module.exports = function (sequelize, DataTypes) {
    var Appointment = sequelize.define('Appointment', {
        id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        appointmentNo: {
            type: DataTypes.STRING(25),
            allowNull: false,
            unique: true
        },
        statusId: {
            type: DataTypes.INTEGER
        },
        timeSlotDetailId: {
            type: DataTypes.INTEGER
        },
        timeSlotDate: {
            type: DataTypes.DATEONLY
        },
        vehicleCustomerId: {
            type: DataTypes.INTEGER
        },
        previousMilleage: {
            type: DataTypes.BIGINT
        },
        currentMilleage: {
            type: DataTypes.BIGINT
        },
        remarks: {
            type: DataTypes.STRING(500) + ' CHARSET utf8 COLLATE utf8_unicode_ci'
        },
        serviceAdvisorId: {
            type: DataTypes.INTEGER
        },
        createdDateTime: {
            type: DataTypes.DATE,
            allowNull: false,
            defaultValue: DataTypes.NOW
        },
        updatedDateTime: {
            type: DataTypes.DATE
        },
        createdBy: {
            type: DataTypes.STRING(150)
        },
        updatedBy: {
            type: DataTypes.STRING(150)
        },
        isDeleted: {
            type: DataTypes.BOOLEAN
        },
    },
        {
            classMethods: {
                associate: function (models) {
                    Appointment.belongsTo(models.VehicleCustomer, { foreignKey: 'vehicleCustomerId', targetKey: 'id' });
                    Appointment.belongsTo(models.TimeSlotDetail, { foreignKey: 'timeSlotDetailId', targetKey: 'id' });
                    Appointment.hasMany(models.RepairOrderMaster, { foreignKey: 'appointmentId', sourceKey: 'id' })
                }
            },
            timestamps: false
        });
    return Appointment;
}
