// tceas-utils/entities/pdModuleType.js
"use strict";

module.exports = function (sequelize, DataTypes) {
    var PDModuleType = sequelize.define('PDModuleType', {
        id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        code: {
            type: DataTypes.STRING(25),
            allowNull: false,
            unique: true
        },
        description: {
            type: DataTypes.TEXT,
            allowNull: false
        },
        status: {
            type: DataTypes.BOOLEAN
        },
        createdDateTime: {
            type: DataTypes.DATE,
            defaultValue: DataTypes.NOW
        },
        updatedDateTime: {
            type: DataTypes.DATE
        },
        createdBy: {
            type: DataTypes.STRING
        },
        modifiedBy: {
            type: DataTypes.STRING
        },
        isDeleted: {
            type: DataTypes.BOOLEAN
        }
    },
        {
            classMethods: {
                associate: function (models) {
                    PDModuleType.hasMany(models.ModuleMovementMatrix, { foreignKey: 'moduleTypeId', sourceKey: 'id' });
                }
            },
            timestamps: false
        });
    return PDModuleType;
}
