// tceas-utils/entities/repairOrderJob.js
"use strict";

module.exports = function (sequelize, DataTypes) {
    var RepairOrderJob = sequelize.define('RepairOrderJob', {
        id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        repairOrderId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        servicePackageJobId: {
            type: DataTypes.INTEGER
        },
        jobId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        pdComeBackJobId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        pdPaymentTypeId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        pdJobSourceId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        pdJobStatusId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        flatRate: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        labourCharge: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        discountPercent: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        subTotal: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        goodWillPercent: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        goodWillAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        netAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
    },
        {
            classMethods: {
                associate: function (models) {
                    RepairOrderJob.hasMany(models.RepairOrderPart, { foreignKey: 'repairOrderJobId', sourceKey: 'id' })
                    RepairOrderJob.belongsTo(models.RepairOrderMaster, { foreignKey: 'repairOrderId', targetKey: 'id' });
                    RepairOrderJob.belongsTo(models.ServicePackageJob, { foreignKey: 'servicePackageJobId', targetKey: 'id' });
                    RepairOrderJob.belongsTo(models.JobMaster, { foreignKey: 'jobId', sourceKey: 'id' });
                    RepairOrderJob.belongsTo(models.PDComeBackJob, { foreignKey: 'pdComeBackJobId', sourceKey: 'id' });
                    RepairOrderJob.belongsTo(models.PDJobStatus, { foreignKey: 'pdJobStatusId', sourceKey: 'id' });
                    RepairOrderJob.belongsTo(models.PDPaymentType, { foreignKey: 'pdPaymentTypeId', sourceKey: 'id' });
                    RepairOrderJob.belongsTo(models.PDJobSource, { foreignKey: 'pdJobSourceId', sourceKey: 'id' });
                    RepairOrderJob.hasOne(models.JobFulfilmentItem, { foreignKey: 'repairOrderJobId', sourceKey: 'id' })
                }
            },
            timestamps: false
        });
    return RepairOrderJob;
}
