// tceas-utils/entities/repairOrderPart.js
"use strict";

module.exports = function (sequelize, DataTypes) {
    var RepairOrderPart = sequelize.define('RepairOrderPart', {
        id: {
            type: DataTypes.INTEGER,
            primaryKey: true,
            autoIncrement: true,
        },
        repairOrderId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        repairOrderJobId: {
            type: DataTypes.INTEGER
        },
        partId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        servicePackagePartId: {
            type: DataTypes.INTEGER
        },
        pdPartTypeId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        pdPaymentTypeId: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        UOM: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        requestQty: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        unitPrice: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        discountPercent: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        discountAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        totalAfterTaxAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        availableQty: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        fullfillQty: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        backOrderId: {
            type: DataTypes.INTEGER,
        },
        subTotal: {
            type: DataTypes.INTEGER,
            allowNull: false
        },
        goodWillPercent: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        goodWillAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        },
        netAmt: {
            type: DataTypes.DECIMAL(18, 2),
            allowNull: false
        }
    },
        {
            classMethods: {
                associate: function (models) {
                    RepairOrderPart.belongsTo(models.RepairOrderMaster, { foreignKey: 'repairOrderId', targetKey: 'id' });
                    RepairOrderPart.belongsTo(models.RepairOrderJob, { foreignKey: 'repairOrderJobId', targetKey: 'id' });
                    RepairOrderPart.belongsTo(models.PartMaster, { foreignKey: 'partId', targetKey: 'id' });
                    RepairOrderPart.belongsTo(models.ServicePackagePart, { foreignKey: 'servicePackagePartId', targetKey: 'id' });
                    RepairOrderPart.belongsTo(models.PDPaymentType, { foreignKey: 'pdPaymentTypeId', targetKey: 'id' });
                    RepairOrderPart.belongsTo(models.PDPartType, { foreignKey: 'pdPartTypeId', targetKey: 'id' });
                    RepairOrderPart.hasMany(models.PartFulfillment, { foreignKey: 'repairOrderPartId', sourceKey: 'id' });
                }
            },
            timestamps: false
        });
    return RepairOrderPart;
}
